keps/sig-network/2112-netpol-namespace-name-selector/README.md

```diff
@@ -0,0 +1,347 @@
+- Allow multiple `namespaceNames` values, so that many namespaces can be 
+allowed using the same field 
+
+### Non-Goals
+
+- Support matching of wildcard namespaces
+- Changing the internal details of how namespaceSelector works with arbitrary syntax
+- Depending on more sophisticated features (like virtual labels) to accomplish 
+this feature without changing the API (note that, if someone were 
+to write a virtualLabels KEP, however, it might change the trajectory of this 
+KEP, which is worth discussing)
+
+## Proposal
+
+In NetworkPolicy specification, inside `NetworkPolicyPeer` specify a new `namespaceNames` field.  
+
+1) One simple way to implement this feature is to modify the NetworkPolicyPeer 
+object, like so.  Since this doesn't involve modifying a go type (i.e. were 
+not replacing a labelSelector with a different type, we expect it 
+to be a cleaner implementation:
+
+```
+type NetworkPolicyPeer struct {
+  // new field
+  namespaceNames []string
+
+  // existing fields...
+	PodSelector *metav1.LabelSelector `json:"podSelector,omitempty" protobuf:"bytes,1,opt,name=podSelector"`
+	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty" protobuf:"bytes,2,opt,name=namespaceSelector"`
+	IPBlock *IPBlock `json:"ipBlock,omitempty" protobuf:"bytes,3,rep,name=ipBlock"`
+}
+```
+
+### User Stories (Optional)
+
+#### Story 1
+
+As an administrator i want to allow access from monitoring pods in kube-system 
+namespace, into any pod in my cluster, as a fundamental policy. 
+
+#### Story 2
+
+As an user I want to "just add" a namespace to my allow list without having to 
+manage the labels which might get added/removed over time from said namespace.
+
+### Notes/Constraints/Caveats (Optional)
+
+### Risks and Mitigations
+
+- CNIs may **NOT** choose, initially to support this construct, and dilligent 
+communication with CNI providers will be needed to make sure its widely 
+adopted.
+- We need to make sure that hidden defaults don't break the meaning of 
+existing policys, for example:
+  - if a namespaece name is retrieved by an api client which doesn't yet 
+  support it, the client doesnt crash (and the plugin doesnt crash, either).
+  - if a namespace name policy doesn't exist (is null), then the policy should 
+  behave identically to any policy made before this field was added, that is, 
+- Both "allow all" and "allow none" could be incorrectly interpreted 
+incorrectly, so we need to make sure that plugins not implementing the feature 
+don't get misled by the new data structure.
+
+The `NetworkPolicyPeer` should be as obvious as possible to interpret in a way that is compatible with the semantics of the v1 policy API **before** this feature was added to it.  
+
+## Design Details
+
+- Add a new selector to the network policy peer data structure which can 
+switch between allowing a "namespaceNames" or "matchLabels" implementation, 
```

Comment:
This seems to imply that a user has to set either `namespaceNames` or `namespaceSelector` but above in the goals it mentions:
```
- Add a `namespaceNames` option (which is additive to the current `matchLabels` selector).
```

Which implies we are taking the union of `namespaceNames` and `namespaceSelector`. Might be good to clarify this a bit.

Reply:
updating this now, I actually thought more about this today